### PR TITLE
Add statuses

### PR DESCRIPTION
**Pull request by VOID404**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: controllers/compassmanager_controller.go [hunk 1]
```diff
@@ -428,3 +306,198 @@ func createCompassRuntimeLabels(kymaLabels map[string]string) map[string]interfa
 
 	return runtimeLabels
 }
+
+type ClusterInterface struct {
```
Comment (mvshao) [inline controllers/compassmanager_controller.go:310]: Compass Manager will have two `cluster` contexts. One for `control-plane` and the other for `SKR`. The name should make it clear what context is being referred to. In my opinion, the keyword `cluster` should be referred to `SKR`. We should change the naming to something more appropriate.

Here it is my proposals:
 - ControlPlaneInterface
 - CentralInterface
 - InfrastructureInterface

What do you think about it?

File: controllers/compassmanager_controller.go [hunk 2]
```diff
@@ -428,3 +306,198 @@ func createCompassRuntimeLabels(kymaLabels map[string]string) map[string]interfa
...
+	labels := make(map[string]string)
+	labels[LabelKymaName] = kyma.Labels[LabelKymaName]
+	labels[LabelCompassID] = compassRuntimeID
+	labels[LabelGlobalAccountID] = kyma.Labels[LabelGlobalAccountID]
+	labels[LabelSubaccountID] = kyma.Labels[LabelSubaccountID]
+	labels[LabelManagedBy] = "compass-manager"
+
+	compassMapping.Labels = labels
+
+	// TODOs add retry for upsert logic
```
Comment (mvshao) [inline controllers/compassmanager_controller.go:445]: Could you delete this comment? It is obsolete, we don't want another retry logic.

File: controllers/compassmanager_controller.go [hunk 3]
```diff
@@ -428,3 +306,198 @@ func createCompassRuntimeLabels(kymaLabels map[string]string) map[string]interfa
...
+	}
+
+	secretList := &corev1.SecretList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		LabelKymaName: name.Name,
+	})
+
+	err := c.kubectl.List(context.TODO(), secretList, &client.ListOptions{
+		LabelSelector: labelSelector,
+		Namespace:     name.Namespace, // TODO: Is this the same namespace?
```
Comment (mvshao) [inline controllers/compassmanager_controller.go:408]: Yes, the `Kubeconfig` is in the same Namespace (kcp-system) as the controller.

File: controllers/compassmanager_controller.go [hunk 4]
```diff
@@ -65,7 +65,7 @@ type Configurator interface {
 }
 
 //go:generate mockery --name=Registrator
```
Comment (koala7659) [inline controllers/compassmanager_controller.go:67]: Please also regenerate mocks for this interface with new name `Registrant`

File: controllers/compassmanager_controller.go (hunk 1 above)
Comment (VOID404) [reply controllers/compassmanager_controller.go:310]: `ControlPlaneInterface` is my favourite out of those.
Or maybe `KCPInterface`?

Comment (mvshao) [reply controllers/compassmanager_controller.go]: KCPInterface is good

File: controllers/compassmanager_controller.go [hunk 5]
```diff
@@ -125,232 +126,124 @@ func (cm *CompassManagerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
...
-	existingMapping.SetLabels(compassMappingLabels)
-	return cm.Client.Update(context.TODO(), &existingMapping)
+	return ctrl.Result{}, nil
 }
 
-func (cm *CompassManagerReconciler) getRuntimeIDFromCompassMapping(kymaName, namespace string) (string, error) {
-	mappingList := &v1beta1.CompassManagerMappingList{}
-	labelSelector := labels.SelectorFromSet(map[string]string{
-		LabelKymaName: kymaName,
-	})
```
Comment (koala7659) [inline controllers/compassmanager_controller.go:263]: Is this comment still relevant? The definition of `setCompassMappingStatus` function is not here. See line 500

**Review by koala7659 — APPROVED** [review]
LGTM

**Pull request merged**